staff/views.py

```diff
@@ -1,48 +1,193 @@
+        else:
+            messages.error(request, 'Teacher User NOT created, your form was not valid, please try again.')
+            return redirect(request, 'staff')
+    form = CreateTeacherForm()
+    return render(request, 'create_teacher_user.html', {'form': form})
+
 
 @login_required
 def create_student_user(request):
-	if Users.objects.filter(email = str(request.user.email)).first().role != 'staff':
-		return HttpResponse('Unauthorized', status=401)
-	return render(request, 'create_student_user.html')
+    if not request.user.groups.filter(name='staff').exists():
+        return HttpResponse('Unauthorized', status=401)
+    if request.method == 'POST':
+        form = CreateStudentForm(request.POST)
+        if form.is_valid():
+            form.save()
+            new_user = create_user_with_profile(form)
+            student_group = Group.objects.get(name='student')
+            student_group.user_set.add(new_user)
+            first_name = form.cleaned_data.get('first_name')
+            messages.success(request, f'Student Account Successfully Created For {first_name}')
+            return redirect('staff')
+        else:
+            messages.error(request, 'Student User NOT created, your form was not valid, please try again.')
+            return redirect(request, 'staff')
+    form = CreateStudentForm()
+    return render(request, 'create_student_user.html', {'form': form})
+
 
 @login_required
 def create_volunteer_user(request):
-	if Users.objects.filter(email = str(request.user.email)).first().role != 'staff':
-		return HttpResponse('Unauthorized', status=401)
-	return render(request, 'create_volunteer_user.html')
+    if not request.user.groups.filter(name='staff').exists():
+        return HttpResponse('Unauthorized', status=401)
+    if request.method == 'POST':
+        form = CreateVolunteerForm(request.POST)
+        if form.is_valid():
+            form.save()
+            new_user = create_user_with_profile(form)
+            volunteer_group = Group.objects.get(name='volunteer')
+            volunteer_group.user_set.add(new_user)
+            first_name = form.cleaned_data.get('first_name')
+            messages.success(request, f'Volunteer Account Successfully Created For {first_name}')
+            return redirect('staff')
+        else:
+            messages.error(request, 'Volunteer User NOT created, your form was not valid, please try again.')
+            return redirect(request, 'staff')
+    form = CreateVolunteerForm()
+    return render(request, 'create_volunteer_user.html', {'form': form})
+
+
+def create_user_with_profile(form):
+    new_user = DjangoUser.objects.create_user(
+        username="%s.%s" % (form.cleaned_data.get('first_name'), form.cleaned_data.get('last_name')),
+        email=form.cleaned_data.get('email'),
+        first_name=form.cleaned_data.get('first_name'),
+        last_name=form.cleaned_data.get('last_name'),
+        password='missionbit'
+    )
```

Comment:
This seems like it might be problematic, with a static password and a username that's not guaranteed to be unique (per django's requirements). Might be worth at least addressing this with an issue or TODO to follow-up on.

Is there a reason we need to know the password? In cases where I've done this sort of thing in the past we would use a randomly generated password that can be set by an admin or with password recovery.

Reply:
Can we discuss further next week?  This is an important usability issue for me and I'd like to get it right.

Reply:
I've noted this issue to be solved before the next PR